### DISCUSSION TITLE
🚀 Config to set allowed directory paths

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#4379] [Feature Requests]
### Elaborating on issues with allowing directories to be created anywhere

This stems from an issue where someone accidentally created a file upload directory in top level public, causing issues- see below.

After thinking on it, adding a config(s) to set allowed and/or disallowed directory paths would be useful.  You can create an upload directory with a path/url anywhere the server allows write access.  But typically, you're going to want all the directories in 'assets' or 'uploads' or... a couple of subfolders.  Being able to set the top levels that directories can be created within makes good sense, and would prevent intended/unintended upload folders being created.

Original discussion details below....

Should we allow file upload directory to upload to root

So in one case {base_url} and {base_path} settings were empty.  In my test, they both pointed to root.

If you go create a file upload directory and don't put in a url or path, those variables are there by default, so there is no validation error.  And now you end up with that directory uploading files to root which is rarely desirable.  And in the first case, they didn't have an index.html file and didn't have php set to be the default file loaded, so they basically brought their homepage down.  The index.html created by making a file directory was now the default content shown when you go to the site.

I'm a bit unsure on what we might want to do here.  It's legit to make that an upload directory if you want to and it's also possible that those base variables point somewhere other than root and you might want the directory there or... IDK.  But I don't love you can do it without some sort of warning.

Either not fill in the fields with the base variables by default, so if they forget to enter something, validation is trigger.  Or more complex, pop a dismissable warning modal or something.

In any case, I thought it was worth discussion.

**Comment by intoeetive** [comment]
For me, the solution I like would be to show confirmation modal if they create upload directory for a dir that already contains the files (so not just root)

But I think this needs some discussion before we implement
